pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -176,91 +174,82 @@ func generateDomainForTargetCPUSetAndTopology(vmi *v1.VirtualMachineInstance, do
+			Name:   f.Name,
+			Policy: f.Policy,
+		})
 	}
-	decoder := xml.NewDecoder(bytes.NewReader(xmlstr))
-	var location = make([]string, 0)
-	var newLocation []string = nil
-	injecting := false
-	for {
-		if newLocation != nil {
-			// Postpone popping end elements from `location` to ensure their removal
-			location = newLocation
-			newLocation = nil
-		}
-		token, err := decoder.RawToken()
-		if err == io.EOF {
-			break
-		}
-		if err != nil {
-			logger.Errorf("error getting token: %v\n", err)
-			return err
-		}
+	return res
+}
 
-		switch v := token.(type) {
-		case xml.StartElement:
-			location = append(location, v.Name.Local)
+func convertKubeVirtCPUTopologyToDomainCPUTopology(cpuTopology *api.CPUTopology) *libvirtxml.DomainCPUTopology {
+	if cpuTopology == nil {
+		return nil
+	}
+	return &libvirtxml.DomainCPUTopology{
+		Sockets: int(cpuTopology.Sockets),
+		Cores:   int(cpuTopology.Cores),
+		Threads: int(cpuTopology.Threads),
+	}
+}
 
-		case xml.EndElement:
-			newLocation = location[:len(location)-1]
+func convertKubeVirtNUMACellToDomainDomainCell(cell []api.NUMACell) []libvirtxml.DomainCell {
+	var ret []libvirtxml.DomainCell
+	for _, c := range cell {
+		v, err := strconv.ParseUint(c.ID, 10, 32)
+		if err != nil {
+			continue
 		}
+		id := uint(v)
+		ret = append(ret, libvirtxml.DomainCell{
+			ID:        &id,
+			CPUs:      c.CPUs,
+			Memory:    uint(c.Memory),
+			Unit:      c.Unit,
+			MemAccess: c.MemoryAccess,
+		})
+	}
+	return ret
+}
 
-		if len(location) >= len(section) &&
-			reflect.DeepEqual(location[:len(section)], section) {
-			injecting = true
-		} else {
-			if injecting == true {
-				// We just left the section block, we're done
-				break
-			} else {
-				// We're not in the section block yet, skipping elements
-				continue
-			}
-		}
+func convertKubeVirtNUMAToDomainNUMA(numa *api.NUMA) *libvirtxml.DomainNuma {
+	if numa == nil {
+		return nil
+	}
+	return &libvirtxml.DomainNuma{
+		Cell: convertKubeVirtNUMACellToDomainDomainCell(numa.Cells),
+	}
+}
 
-		if injecting {
-			if err := encoder.EncodeToken(xml.CopyToken(token)); err != nil {
-				logger.Reason(err).Errorf("Failed to encode token %v", token)
-				return err
-			}
-		}
+func convertKubeVirtCPUToDomainCPU(cpu *api.CPU) *libvirtxml.DomainCPU {
+	if cpu == nil {
+		return nil
+	}
+	return &libvirtxml.DomainCPU{
+		Mode: cpu.Mode,
+		Model: &libvirtxml.DomainCPUModel{
+			Value: cpu.Model,
+		},
+		Features: convertKubeVirtCPUFeatureToDomainCPUFeature(cpu.Features),
+		Topology: convertKubeVirtCPUTopologyToDomainCPUTopology(cpu.Topology),
+		Numa:     convertKubeVirtNUMAToDomainNUMA(cpu.NUMA),
 	}
 
-	return nil
 }
 
-// This function returns true for every section that should be adjusted with target data when migrating a VMI
-//
-//	that includes dedicated CPUs
-//
-// Strict mode only returns true if we just entered the block
-func shouldOverrideForDedicatedCPUTarget(section []string, strict bool) bool {
```

Comment:
This logic was removed but not sure we still have it with the patched code.

Reply:
It shouldn't as it was used for understaing if we were in the right block of the XML. Now, this part is replaced by the Marshal function from libvirtxml